py-polars/tests/unit/datatypes/test_struct.py

```diff
@@ -967,3 +967,19 @@ def test_struct_name_passed_in_agg_apply() -> None:
             ]
         ],
     }
+
+
+def test_struct_lit_cast() -> None:
+    df = pl.DataFrame({"a": [1, 2, 3]})
+    schema = {"a": pl.Int64, "b": pl.List(pl.Int64)}
+
+    for lit in [pl.lit(None), pl.lit([[]])]:
+        s = df.select(pl.struct([pl.col("a"), lit.alias("b")], schema=schema))["a"]  # type: ignore[arg-type]
```

Comment:
@stijndegooijer I don't understand why this one fails, maybe you'd care to take a look sometime?

Reply:
You mean the mypy error? I'll take a look!

Reply:
@stinodego 

(this is my work account 😄 )

Reply:
Which one should I refer to?

Reply:
Tag @stinodego if you need me for something!

Reply:
@ritchie46 There isn't really anything to fix here, it's just some limitations of mypy. Conside the example below:

```python
d = {"a": 1, "c": "hello"}

def print_dict(d: dict[str, int | str]) -> None:
    print(d)

print_dict(d)

```
Running mypy results in:
```
error: Argument 1 to "print_dict" has incompatible type "Dict[str, object]"; expected "Dict[str, Union[int, str]]"  [arg-type]
```

mypy is unable to generalize the dict into a type of `Dict[str, Union[int, str]]`. It infers a more generic `Dict[str, object]`, which of course does not fit the type signature of the function. _(Interestingly enough, this works fine when combining `int` and `float`)._

It can be solved by explicitly casting the dictionary (`d: Dict[str, Union[int, str] = {...}`)to the expected type, or using a `# type: ignore` comment.